### PR TITLE
added harvester exceptions / some comments

### PR DESCRIPTION
**Pull request by Pentadrago**
* 3 new harvester exceptions instead of generic ones
 * JobTypeAlreadyRegisteredException
 * NoSuchJobTypeException
 * JobAlreadyScheduledException
* better javadoc

### CODE REVIEW

File: harvester/src/main/java/com/github/reflectoring/infiniboard/harvester/scheduling/SchedulingService.java [hunk 1]
```diff
@@ -89,22 +105,31 @@ private JobDataMap createContextData() {
...
-            LOG.error("no job of type {} was found", type);
-            return;
+            throw new NoSuchJobTypeException(String.format("no job of type %s was found", type));
         }
 
         if (checkJobExists(name, group)) {
-            LOG.error("job of type {}, group {}, and name{} already exists {} was found", type, group, name);
-            throw new SchedulerException("Job already exists");
+            throw new JobAlreadyScheduledException(
+                    String.format("job of type %s, group %s, and name%s already exists, could not be scheduled", type,
```
Comment (matthiasbalke) [inline harvester/src/main/java/com/github/reflectoring/infiniboard/harvester/scheduling/SchedulingService.java]: missing space

Comment (matthiasbalke) [inline harvester/src/main/java/com/github/reflectoring/infiniboard/harvester/scheduling/SchedulingService.java]: please add single quotes around placeholders e.g. `'%s'`

File: harvester/src/main/java/com/github/reflectoring/infiniboard/harvester/source/SourceJob.java [hunk 2]
```diff
@@ -36,5 +36,11 @@ public void execute(JobExecutionContext context)
         }
     }
 
+    /**
+     * called after checking if the job could be run (the associated widget exists)
```
Comment (matthiasbalke) [inline harvester/src/main/java/com/github/reflectoring/infiniboard/harvester/source/SourceJob.java]: please format the javadoc

File: harvester/src/test/java/com/github/reflectoring/infiniboard/harvester/scheduling/SchedulingServiceTest.java [hunk 3]
```diff
@@ -54,21 +52,24 @@ public void setup()
         schedulingService.registerJob(TEST_JOB, TestJob.class);
     }
 
-    @Test
+    @Test(expected = JobTypeAlreadyRegisteredException.class)
```
Comment (matthiasbalke) [inline harvester/src/test/java/com/github/reflectoring/infiniboard/harvester/scheduling/SchedulingServiceTest.java]: prefer using ExpectedException over @Test expected

Comment (Pentadrago) [inline harvester/src/test/java/com/github/reflectoring/infiniboard/harvester/scheduling/SchedulingServiceTest.java]: agreed

File: harvester/src/main/java/com/github/reflectoring/infiniboard/harvester/source/SourceJob.java (hunk 2 above)
Comment (Pentadrago) [inline harvester/src/main/java/com/github/reflectoring/infiniboard/harvester/source/SourceJob.java]: done

File: harvester/src/main/java/com/github/reflectoring/infiniboard/harvester/scheduling/SchedulingService.java [hunk 4]
```diff
@@ -120,7 +146,7 @@ public void scheduleJob(String group, SourceConfig config)
                                       .repeatForever())
                 .build();
         scheduler.scheduleJob(job, trigger);
-        LOG.debug("scheduled job ({}:{}) of type {} for {} ms", group, config.getId(), config.getType(),
+        LOG.debug("scheduled job ({}:{}) of type '{}' for {} ms", group, config.getId(), config.getType(),
```
Comment (matthiasbalke) [inline harvester/src/main/java/com/github/reflectoring/infiniboard/harvester/scheduling/SchedulingService.java]: missing quotes

File: harvester/src/main/java/com/github/reflectoring/infiniboard/harvester/scheduling/SchedulingService.java (hunk 1 above)
Comment (Pentadrago) [inline harvester/src/main/java/com/github/reflectoring/infiniboard/harvester/scheduling/SchedulingService.java]: done (including other exception and log messages in harvester module)

**Pull request merged**